### ISSUE TITLE
Cannot run Android release build when index.android.js not in main directory

### ISSUE THREAD

**Issue by dannycochran** [facebook/react-native#11571]
### Description

I recently upgraded to react-native 0.39.2 from 0.37, and my command to run a production build on the emulator no longer works.

`
react-native run-android --configuration Release --variant prod --root artifacts/lib
`

Fails with the error:

`
Cannot find entry file index.android.js in any of the roots: ["/Users/Documents/reactApp"]
`

I'm using Typescript to compile assets into an artifacts folder, such that my directory structure is:

- reactApp
  - package.json
  - ios
  - android
    - gradle
    - ...
  - artifacts
    - lib
      - index.android.js
      - ...
    - ...
  - src
    - lib
      - index.android.ts
      - ...
    - ...

### Reproduction

Have a directory structure similar to the one above where index.android.js is not in the main directory folder and run:
`
react-native run-android --configuration Release --variant prod
`
### Solution

I believe this was caused by a pull request in 0.39, from @SandroMachado 

https://github.com/facebook/react-native/commit/2a2d3c6e5a54c553a5c8a9932d8204463e7bbb49#diff-a003f89d34db18b4567fdb55080a003eR103

The react-native bundle command seems to assume the index.android.js exists in the root directory of a project, and no longer sets the process current directory to be the root arg passed in when running react-native run-android.

### Additional Information

* React Native version: 0.39.2
* Platform: Android
* Operating System: MacOS

**Comment by SandroMachado** [comment]
Use `--configuration` and not `config`.

**Comment by dannycochran** [comment]
@SandroMachado sorry, my mistake. I am using --configuration.

**Comment by SandroMachado** [comment]
This is the command executed to generate the release `react-native bundle --platform android --dev false --entry-file index.android.js --bundle-output ${androidProjectDir}/app/src/main/assets/index.android.bundle --assets-dest ${androidProjectDir}/app/src/main/res/`

So the problem is your `index.android.js` is under the `artifacts` directory?

**Comment by dannycochran** [comment]
correct, index.android.js is under artifacts/lib.

when running in dev, I run two separate commands on two different terminal screens:

screen 1:
`
node node_modules/react-native/local-cli/cli.js start --root artifacts/lib --reset-cache
`
screen 2:
`
react-native run-android --variant dev
`

In production, I run:

`
react-native run-android --configuration Release --variant prod --root artifacts/lib
`

But the artifacts/lib root isn't respected for the line you pasted above, [line 110 of local-cli/runAndroid/runAndroid.js](https://github.com/facebook/react-native/commit/2a2d3c6e5a54c553a5c8a9932d8204463e7bbb49#diff-a003f89d34db18b4567fdb55080a003eR110)

**Comment by SandroMachado** [comment]
Yup, it is an issue. It is easy to fix, unfortunately that code is not on `master` anymore. It was removed by https://github.com/facebook/react-native/commit/85c8333bc87d27a7f5f31a74d62c3d0b3ad631f9 So I cannot make a PR to fix it...

**Comment by dannycochran** [comment]
dang. what would the fix look like? I hardcoded line 115 to look like this:

`
'--entry-file artifacts/lib/index.android.js ' +
`

and that results in a new error that makes little sense to me:

```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:processProdReleaseResources'.
> com.android.ide.common.process.ProcessException: org.gradle.process.internal.ExecException: Process 'command '/Users/dannycochran/Library/Android/sdk/build-tools/23.0.1/aapt'' finished with non-zero exit value 1

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED
```

**Comment by SandroMachado** [comment]
Hum, can this be a memory (RAM) issue? In the past, developing for Android, I saw that kind of errors causes by insufficient memory.

**Comment by dannycochran** [comment]
I'm guessing this is a separate issue from the one I originally posted, and may have to do with the fact that my app has a webview which loads various assets. When I run:

`
react-native run-android --configuration Release --variant prod
`

Five additional directories are created in android/app/src/main/res/:

- drawable-hdpi
- drawable-mdpi
- drawable-xhdpi
- drawable-xxhdpi
- drawable-xxxhdpi

The bundler also creates two files in android/app/src/main/assets/:

- index.android.bundle
- index.android.bundle.meta

If I remove all these files right after they're created (git clean -f -d) by the bundle command, the app builds successfully on the emulator. If I don't remove them, I get the error above.

**Comment by dannycochran** [comment]
this was fixed by 0.41.0

**Comment by sahir** [comment]
Thanks @dannycochran it's fixed in 0.41.0. This is the command executed to generate the release

react-native bundle --platform android --dev false --entry-file index.android.js \  --bundle-output ../android/app/src/main/assets/index.android.bundle \--assets-dest ../android/app/src/main/res/

**Comment by erm3nda** [comment]
Case of my install i have modified the @sahir 's command to:

`react-native bundle --platform android --dev false --entry-file index.android.js --bundle-output android/app/src/main/assets/index.android.bundle --assets-dest android/app/src/main/res/`

Now the bundle file is present.
